src/main/java/frc/robot/Robot.java

```diff
@@ -10,114 +10,98 @@
-  // auto classes
-  private VisionControl visionControl;
-  private Vision vision;
-  private VisionData vData;
+    private OperatorInterface oi = new OperatorInterface();
+    // auto classes
+    private Buttons buttons = new Buttons(oi);
+    private VisionControl visionControl;
+    private Vision        vision;
+    private VisionData    vData;
 
-  /**
-   * This function is run when the robot is first started up and should be used
-   * for any
-   * initialization code.
-   */
-  @Override
-  public void robotInit() {
-  }
+    /**
+     * This function is run when the robot is first started up and should be
+     * used for any initialization code.
+     */
+    @Override
+    public void robotInit() {}
 
-  /**
-   * This function is called every robot packet, no matter the mode. Use this for
-   * items like
-   * diagnostics that you want ran during disabled, autonomous, teleoperated and
-   * test.
-   *
-   * <p>
-   * This runs after the mode specific periodic functions, but before LiveWindow
-   * and
-   * SmartDashboard integrated updating.
-   */
-  @Override
-  public void robotPeriodic() {
-  }
+    /**
+     * This function is called every robot packet, no matter the mode. Use this
+     * for items like diagnostics that you want ran during disabled, autonomous,
+     * teleoperated and test.
+     * <p>
+     * This runs after the mode specific periodic functions, but before
+     * LiveWindow and SmartDashboard integrated updating.
+     */
+    @Override
+    public void robotPeriodic() {}
 
-  /**
-   * This autonomous (along with the chooser code above) shows how to select
-   * between different
-   * autonomous modes using the dashboard. The sendable chooser code works with
-   * the Java
-   * SmartDashboard. If you prefer the LabVIEW Dashboard, remove all of the
-   * chooser code and
-   * uncomment the getString line to get the auto name from the text box below the
-   * Gyro
-   *
-   * <p>
-   * You can add additional auto modes by adding additional comparisons to the
-   * switch structure
-   * below with additional strings. If using the SendableChooser make sure to add
-   * them to the
-   * chooser code above as well.
-   */
-  @Override
-  public void autonomousInit() {
-    initialize();
-  }
+    /**
+     * This autonomous (along with the chooser code above) shows how to select
+     * between different autonomous modes using the dashboard. The sendable
+     * chooser code works with the Java SmartDashboard. If you prefer the
+     * LabVIEW Dashboard, remove all of the chooser code and uncomment the
+     * getString line to get the auto name from the text box below the Gyro
+     * <p>
+     * You can add additional auto modes by adding additional comparisons to the
+     * switch structure below with additional strings. If using the
+     * SendableChooser make sure to add them to the chooser code above as well.
+     */
+    @Override
+    public void autonomousInit() {
+        initialize();
```

Comment:
I wonder if we should do initialization the usual way (in constructors). We did this when we had feature flags set table at runtime via the dashboard. If we only set the at compile time, then we don't need to defer initialization anymore.

Reply:
True, ill leave initialisation as a thing but it will only be called in robotinit unless we want to add back the dashboard stuff